File: client/src/services/semesterService.js
import axiosClient from "./axiosClient";

const baseUrl = "/semester";
const semesterService = {
  fetchAllSemesters: (pageNumber, itemsPerPage, sortBy) => {
    const requestUrl = `${baseUrl}?pageNumber=${pageNumber}&pageSize=${itemsPerPage}&sortBy=${sortBy}`;
    return axiosClient.get(requestUrl);
  },
  createSemester: (data) => {
    const requestUrl = `${baseUrl}`;
    return axiosClient.post(requestUrl, data);
  },
  updateSemester: (id, data) => {
    const requestUrl = `${baseUrl}/${id}`;
    return axiosClient.put(requestUrl, data);
  },
};

export default semesterService;

File: server/src/main/java/com/bosch/topicregistration/api/semester/SemesterService.java
package com.bosch.topicregistration.api.semester;

import com.bosch.topicregistration.api.response.PageResponse;
import com.bosch.topicregistration.api.response.Response;

import java.util.List;

public interface SemesterService {
    Semester getActivatedSemester();

    Response<PageResponse<List<SemesterDTO>>> getListSemester(Integer pageNumber, Integer pageSize, String sortBy);

    Response<Void> createSemester(SemesterRequest request);

    Response<Void> updateSemester(String semesterId, SemesterRequest request);
}
